site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/SiteStore.java

```diff
@@ -39,19 +39,30 @@ public SiteStore(FhirRepository repository,
      */
     @Override
     public String saveEntity(Site entity) {
+        String orgId = "";
 
-        // TODO: Check if the Organization already exists.
+        //Check if the Organization already exists.
+        String orgIdFound = repository.findOrganizationByName(entity.getName());
 
-        Organization org = converter.convert(entity);
-        String orgId = repository.saveOrganization(org);
-        org.setId(orgId);
-        logger.info(FhirRepo.SAVE_REQUEST.message(), orgId);
+        if (FhirRepo.NOT_FOUND.message().equalsIgnoreCase(orgIdFound)) {
```

Comment:
As commented above in the method - there're better ways to check if you found the org or not.

Reply:
This has been addressed